packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  Animated,
+  Easing,
+  Text,
+  TextStyle,
+  View,
+  ViewStyle,
+} from 'react-native'
+import React, { useEffect } from 'react'
+
+import { Colors } from '@department-of-veterans-affairs/mobile-tokens'
+import { Icon, IconProps } from '../Icon/Icon'
+import { Spacer } from '../Spacer/Spacer'
+import { useColorScheme } from '../../utils'
+
+export type LoadingIndicatorProps = {
+  /** AccessibilityLabel for the text */
+  a11yLabel?: string
+} & (
+  | {
+      /** Text appearing below indicator. Required if no children included */
+      text: string
+      /** Optional custom content to display below indicator */
+      children?: React.ReactNode
+    }
+  | {
+      /** Text appearing below indicator. Optional if children included */
+      text?: string
+      /** Optional custom content to display below indicator */
+      children: React.ReactNode
+    }
+)
+
+export const LoadingIndicator: React.FC<LoadingIndicatorProps> = ({
+  text,
+  a11yLabel,
+  children,
+}) => {
+  const colorScheme = useColorScheme()
+  const isDarkMode = colorScheme === 'dark'
+  const rotation = new Animated.Value(0)
+
+  useEffect(() => {
+    const animate = () => {
+      rotation.setValue(0) // Reset the rotation value to 0
+      Animated.timing(rotation, {
+        toValue: 1,
+        duration: 1500,
+        easing: Easing.linear,
+        useNativeDriver: true,
+      }).start(() => animate()) // Loop the animation
+    }
+    animate()
+  }, [rotation])
+
+  const rotate = rotation.interpolate({
+    inputRange: [0, 1],
+    outputRange: ['0deg', '360deg'],
+  })
+
+  const containerStyle: ViewStyle = {
+    width: '100%',
+    alignItems: 'center',
+  }
+
+  const indicatorStyle: ViewStyle = {
+    width: 50,
+    height: 50,
+    transform: [{ rotate }],
+    justifyContent: 'center',
+    alignItems: 'center',
+  }
+
+  const iconProps: IconProps = {
+    name: 'LoadingIndicator',
+    width: 50,
+    height: 50,
+    fill: isDarkMode ? Colors.uswdsBlueVivid30 : Colors.primary,
```

Comment:
Looking at Figma, I think we now have semantic tokens to use for these--this is set to `vads-color-action-foreground-default`.

Reply:
Will need to merge [@jessicawoodin's PR first](https://github.com/department-of-veterans-affairs/va-mobile-library/pull/349) but I agree

Reply:
Not certain, but in theory your branch updating to the same version hers does shouldn't cause conflict if it moved first. Planning to look at her PR next though.

Reply:
Updated tokens but linting/tests will fail for other components while her PR is unmerged